### ISSUE TITLE
Index existing WARC content

### ISSUE THREAD

**Issue by edsu** [sul-dlss/was-pywb#13]
We need to index the entire set of WARC data in `/web-archiving-stacks/data/collections` using [cdxj-indexer](https://github.com/webrecorder/cdxj-indexer/). There is a sloppy start at doing this in parallel in `bin/index.rb` which can be adapted or we can start over with a new program.

`index.rb` should generate a CDXJ file for every WARC file, ideally in parallel to use the available CPUs. Once complete these individual WARC files need to be merged into a single CDXJ file named `index.cdxj` using `sort --merge` similar to what is happening in [was_robots_suite](https://github.com/sul-dlss/was_robot_suite/blob/main/lib/was_crawl_dissemination/cdx_merge_sort_publish_service.rb#L19-L23).

It should be possible to rerun `bin/index.rb` to pick up new WARC data that is acquired during the workcycle. One approach to doing this is to leave the individual CDXJ files on disk so that they don't need to be regenerated, and so they can be remerged to created the single `index.cdxj`.

The `was-dev` server has a read-only copy of `/web-archiving-stacks/` which can be used.

**Comment by edsu** [comment]
Yes, I think i mentioned that at the end?

**Comment by edsu** [comment]
Indexing is running again, it looks like it will be done in 72 hours. This will stay open until it is complete.

**Comment by edsu** [comment]
When reindexing WARC files that failed on the first pass it ran into a swathe of some huge WARC files whose CDXJ files exceed working memory (8GB on was-dev). For example:

```
-rw-r--r-- 1 was was 89G Sep 30  2020 /web-archiving-stacks/data/collections/bs665fm2060/xc/966/bm/1153/ARCHIVEIT-924-SEMIANNUAL-JOB509080-20171206025430169-00005.warc.gz
```

When given the `--sort` option cdxj-indexer will keep the CDXJ index in memory and will sort it at the end. I think we may need to write the CDXJ to disk unsorted and then sort with the `sort` command line utility which can use disk as needed.

**Comment by edsu** [comment]
It turns out that the real culprit here might actually be --post-append. This gobbles up all available memory on was-dev:

```
$ cdxj-indexer --post-append /web-archiving-stacks/data/collections/bs665fm2060/xc/966/bm/1153/ARCHIVEIT-924-SEMIANNUAL-JOB509080-20171206025430169-00005.warc.gz > /dev/null
```

I've ticketed this over in https://github.com/webrecorder/cdxj-indexer/issues/17

**Comment by edsu** [comment]
The first run at this finished. It took about 5 days to do the cdxj indexing and then to concatenate and sort them. We may want to redo it before launch to make sure it is up to date.
